vumi/transports/dmark/dmark_ussd.py

```diff
@@ -211,3 +207,53 @@ def handle_outbound_message(self, message):
                 sent_message_id=message['message_id'],
                 reason="Could not find original request.")
             returnValue(nack)
+
+    def on_down_response_time(self, message_id, time):
+        request = self.get_request(message_id)
+        # We send different status events for error responses
+        if request.code < 200 or request.code >= 300:
+            return
+        return self.add_status(
+            component='response',
+            status='down',
+            type='slow_response',
+            message='Very slow response',
+            reasons=[
+                'Response took longer than %fs' % (
+                    self.response_time_down,)
+            ],
+            details={
+                'response_time': time,
+            })
+
+    def on_degraded_response_time(self, message_id, time):
+        request = self.get_request(message_id)
+        # We send different status events for error responses
+        if request.code < 200 or request.code >= 300:
+            return
+        return self.add_status(
+            component='response',
+            status='degraded',
+            type='slow_response',
```

Comment:
Wondering if this should have a different `type` to the down status event? They seem like different types.